scalikejdbc-streams/src/test/scala/somewhere/DatabasePublisherSpec.scala

```diff
@@ -34,102 +35,247 @@ class DatabasePublisherSpec
 
   behavior of "DatabasePublisher"
 
+  // ------------------------------------------
+  // SyncSubscriber
+  // ------------------------------------------
+
   it should "be subscribed by SyncSubscriber" in {
-    val promise = Promise[Int]()
-    val subscriber = new SyncSubscriber[Int] {
-      private[this] var previousElement = 0
+    val publisher: DatabasePublisher[Int] = DB readOnlyStream {
+      SQL(s"select id from $tableName").map(r => r.int("id")).iterator
+    }
+
+    val consumedCountPromise: Promise[Int] = Promise[Int]()
+    val subscriber: SyncSubscriber[Int] = new SyncSubscriber[Int] {
+      private[this] val consumedCount = new AtomicInteger(0)
 
       override def foreach(element: Int): Boolean = {
-        log.debug(s"whenNext element=$element")
-        if (element < previousElement) {
-          false
-        } else {
-          previousElement = element
-          true
-        }
+        val consumed = consumedCount.incrementAndGet()
```

Comment:
Here we tested that elements will arrive in order.
(Also, since SyncSubscriber and [AsyncSubscriber](https://github.com/reactive-streams/reactive-streams-jvm/blob/master/examples/src/main/java/org/reactivestreams/example/unicast/AsyncSubscriber.java#L187) would obeying Spec rule 2.7, I do not think you need to use AtomicInteger.)
How is this?

```scala
val answer = (1 to 100)

val m = scala.collection.mutable.Seq.newBuilder[Int]
override def foreach(element: Int): Boolean = {
  m += element
  true
}

assert(m == answer)
```